src/Client.php

```diff
@@ -104,6 +105,10 @@ public function ingestText(
         ?array $metadata = null,
         ?string $reference = null
     ): IngestionResult {
+        if (is_array($metadata)) {
+            $metadata = (new Metadata())
+                ->setCollection($metadata);
+        }
```

Comment:
We can avoid `if (...)` if we make the `::setCollection()` parameter nullable. Then we convert a potential null into an empty array, inside the method.